packages/main/src/DatePicker.js

```diff
@@ -479,17 +479,20 @@ class DatePicker extends DateComponentBase {
 
 		const modifiedDate = modifyDateBy(CalendarDate.fromLocalJSDate(this.dateValue), amount, unit, this._minDate, this._maxDate);
 		const newValue = this.formatValue(modifiedDate.toUTCJSDate());
-		this._updateValueAndFireEvents(newValue, true, ["change", "value-changed"]);
+		this._updateValueAndFireEvents(newValue, true, ["change", "value-changed"], true);
 	}
 
-	_updateValueAndFireEvents(value, normalizeValue, events) {
+	_updateValueAndFireEvents(value, normalizeValue, events, shouldChangeValueState) {
 		const valid = this._checkValueValidity(value);
 		if (valid && normalizeValue) {
 			value = this.normalizeValue(value); // transform valid values (in any format) to the correct format
 		}
-
+		
 		this.value = value;
```

Comment:
Not sure if the `value `property should be updated, when `shouldChangeValueState `argument is `false`.
The sap.m.DatePicker control doesn't react in such case for comparison.